scanpy/tools/_louvain.py

```diff
@@ -133,7 +137,22 @@ def louvain(
                 g, partition_type,
                 **partition_kwargs,
             )
-            # adata.uns['louvain_quality'] = part.quality()
+            if 'quality' in dir(part):
+                q = part.quality()
+                if isinstance(part, louvain.RBConfigurationVertexPartition) and q > 1.0: #scale
+                    m = weights.sum() if use_weights else g.ecount()
+                    q /= m if directed else m*2
+                if isinstance(part, (louvain.RBConfigurationVertexPartition,
+                                     louvain.ModularityVertexPartition)):
+                    qual_type = ' (scaled modularity)'
+                else:
+                    qual_type = ''
+                adata.uns[uns_key]['quality'] = q
+                quality_msg = (
+                    f'\n'
+                    f'    quality of the partitioning{qual_type} is {q:.3f}\n'
+                    f'    added "quality" key to adata.uns["{uns_key}"]'
+                )
```

Comment:
This block of code should be exactly the same as in the _leiden module. Please create a helper function instead of copying and pasting code.

Reply:
leiden and louvain modules are over 90% the same. So making this a helper would be just deferring the inevitable merger of the two. So I'd like to put the same effort to merge them instead of writing a helper function just for the partition quality.

How does that sound?

Reply:
Related issue: https://github.com/theislab/scanpy/issues/570

Reply:
sounds great!